compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/TooltipArea.desktop.kt

```diff
@@ -204,9 +209,23 @@ interface TooltipPlacement {
     /**
      * Returns [PopupPositionProvider] implementation.
      */
+    @Deprecated(
+        message = "Use the overload that takes a cursor position; will be removed in Compose 1.5"
+    )
     @Composable
     fun positionProvider(): PopupPositionProvider
 
+    /**
+     * Returns [PopupPositionProvider] implementation.
+     *
+     * @param cursorPosition The position of the mouse cursor relative to the tooltip area.
+     */
+    @Suppress("DEPRECATION")
+    @Composable
+    fun positionProvider(cursorPosition: IntOffset): PopupPositionProvider {
```

Comment:
We can pass Offset here instead of IntOffset, WDYT?

Reply:
+1 to the question

Reply:
This question was arised in the past, and we decided that we shouldn't be opinionated about how the platforms sends us pointer coordinates. So it should be `Offset`.

Modern desktop platforms, indeed send them as Int, but the other platforms send them as Float (iOS, Android), and we can imagine a new hypothetical desktop platform that also sends them as Float's and do some stabilization to avoid fluctuations.

Reply:
So all APIs that relate to pointer positions should use `Offset` and convert to `Int` only at the last moment?

Reply:
Yes. 

> convert to Int only at the last moment?

I would say at the moment when we do layouting, because it should manipulate round pixels to avoid fuzzy rendering or alignment errors

Reply:
I changed all the positions to be `Offset` and only `PopupPositionProviderAtOffset` converts them to pixels.